src/stretchy.js

```diff
@@ -137,309 +92,52 @@ const katexImagesData = {
-h399450v120H429l-6-1c-124.688-8-235-61.7-331-161C60.687 138.7 32.312 99.3 7 54
-L0 41V6z'/></svg><svg x='25%' width='50%' viewBox='0 0 400000 548'
-preserveAspectRatio='xMidYMin slice'><path d='M199572 214
-c100.7 8.3 195.3 44 280 108 55.3 42 101.7 93 139 153l9 14c2.7-4 5.7-8.7 9-14
- 53.3-86.7 123.7-153 211-199 66.7-36 137.3-56.3 212-62h199568v120H200432c-178.3
- 11.7-311.7 78.3-403 201-6 8-9.7 12-11 12-.7.7-6.7 1-18 1s-17.3-.3-18-1c-1.3 0
--5-4-11-12-44.7-59.3-101.3-106.3-170-141s-145.3-54.3-229-60H0V214z'/></svg>
-<svg x='74.9%' width='25.1%' viewBox='0 0 400000 548'
-preserveAspectRatio='xMaxYMin slice'><path d='M399994 0l6 6
-v35l-6 11c-56 104-135.3 181.3-238 232-57.3 28.7-117 45-179 50H-300V214h399897
-c43.3-7 81-15 113-26 100.7-33 179.7-91 237-174 2.7-5 6-9 10-13 .7-1 7.3-1 20-1
-h17z'/></svg>`,
-
-    // undergroup is from the MnSymbol package (public domain)
-    undergroup: `><svg width='50.1%' viewBox='0 0 400000 342'
-preserveAspectRatio='xMinYMin slice'><path d='M400000 262
-H435C64 262 168.3 112.6 21 82c-5.9-1.2-18 0-18 0-2 0-3 1-3 3v38c76 158 257 219
- 435 219h399565z'/></svg><svg x='50%' width='50%' viewBox='0 0 400000 342'
-preserveAspectRatio='xMaxYMin slice'><path d='M0 262h399565
-c371 0 266.7-149.4 414-180 5.9-1.2 18 0 18 0 2 0 3 1 3 3v38c-76 158-257
- 219-435 219H0z'/></svg>`,
-
-    // widehat1 is a modified version of a glyph from the MnSymbol package
-    widehat1: ` viewBox='0 0 1062 239' preserveAspectRatio='none'>
-<path d='M529 0h5l519 115c5 1 9 5 9 10 0 1-1 2-1 3l-4 22
-c-1 5-5 9-11 9h-2L532 67 19 159h-2c-5 0-9-4-11-9l-5-22c-1-6 2-12 8-13z'/>`,
-
-    // Ditto widehat2, widehat3, and widehat4
-    widehat2: ` viewBox='0 0 2364 300' preserveAspectRatio='none'>
-<path d='M1181 0h2l1171 176c6 0 10 5 10 11l-2 23c-1 6-5 10
--11 10h-1L1182 67 15 220h-1c-6 0-10-4-11-10l-2-23c-1-6 4-11 10-11z'/>`,
-
-    widehat3: ` viewBox='0 0 2364 360' preserveAspectRatio='none'>
-<path d='M1181 0h2l1171 236c6 0 10 5 10 11l-2 23c-1 6-5 10
--11 10h-1L1182 67 15 280h-1c-6 0-10-4-11-10l-2-23c-1-6 4-11 10-11z'/>`,
-
-    widehat4: ` viewBox='0 0 2364 420' preserveAspectRatio='none'>
-<path d='M1181 0h2l1171 296c6 0 10 5 10 11l-2 23c-1 6-5 10
--11 10h-1L1182 67 15 340h-1c-6 0-10-4-11-10l-2-23c-1-6 4-11 10-11z'/>`,
+// In the katexImagesData object just below, the dimensions all
+// correspond to path geometry inside the relevant SVG.
+// For example, \overrightarrow uses the same arrowhead as glyph U+2192
+// from the KaTeX Main font. The scaling factor is 1000.
+// That is, inside the font, that arrowhead is 522 units tall, which
+// corresponds to 0.522 em inside the document.
 
-    xcancel: `<line x1='0' y1='0' x2='100%' y2='100%' stroke-width='0.046em'/>
-<line x1='0' y1='100%' x2='100%' y2='0' stroke-width='0.046em'/>`,
+const katexImagesData = {
+                   //   path(s), minWidth, height, align
+    overrightarrow: [["rightarrow"], 0.888, 522, "xMaxYMin"],
```

Comment:
Why has the `minWidth` changed for some of these?  Previously it was `0.5` for `overrightarrow`.

Reply:
I discovered `opentype.js`. It made it easier to quickly look at the width of each glyph. So instead of using the width from one glyph and applying it several times, I got a number that matched that specific glyph. For instance, `\overrightarrow` is exactly 888 units width inside the font.

Reply:
Awesome!  Are there situations in which the different `minWidth` will have an impact on the rendering?

Reply:
A function with an empty argument, say `\overrightarrow{}` will be rendered with the min-width. I suppose that min-width might also come into play if the argument is a single, narrow letter like `\imath`.